Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/api/settings/FacetedSearch/FacetedSearchClassificationResource.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * The Apereo Foundation licenses this file to you under the Apache License,
+ * Version 2.0, (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tle.web.api.settings.FacetedSearch
+
+import java.util.Date
+
+import com.tle.common.Check
+import com.tle.common.institution.CurrentInstitution
+import com.tle.core.facetedsearch.bean.FacetedSearchClassification
+import com.tle.legacy.LegacyGuice
+import com.tle.web.api.ApiErrorResponse
+import io.swagger.annotations.{Api, ApiOperation, ApiParam}
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.Response.Status
+import javax.ws.rs.{DELETE, GET, POST, PUT, Path, PathParam, Produces}
+import org.jboss.resteasy.annotations.cache.NoCache
+
+import scala.collection.mutable.ArrayBuffer
+
+@NoCache
+@Path("settings/")
+@Produces(value = Array("application/json"))
+@Api(value = "Settings")
+class FacetedSearchClassificationResource {
+  val service          = LegacyGuice.facetedSearchClassificationService
+  val privilegeChecker = LegacyGuice.searchPrivProvider
+
+  @GET
+  @Path("facetedsearch/classification")
+  @ApiOperation(
+    value = "List faceted search classifications",
+    notes = "This endpoint is used to retrieve faceted search classifications.",
+    response = classOf[FacetedSearchClassification],
+    responseContainer = "List"
+  )
+  def getAll: Response = {
+    privilegeChecker.checkAuthorised()
+    val classifications = service.enumerateAll()
+    Response.ok().entity(classifications).build()
+  }
+
+  @GET
+  @Path("facetedsearch/classification/{id}")
+  @ApiOperation(
+    value = "Retrieve one faceted search classification",
+    notes = "This endpoint is used to retrieve one faceted search classification.",
+    response = classOf[FacetedSearchClassification]
+  )
+  def get(@ApiParam(value = "ID") @PathParam("id") id: Long): Response = {
```

Comment:
Isn't `value` redundant?

```suggestion
  def get(@ApiParam("ID") @PathParam("id") id: Long): Response = {
```